operator/pkg/translate/translate.go

```diff
@@ -583,6 +587,25 @@ func (t *Translator) TranslateHelmValues(iop *v1alpha1.IstioOperatorSpec, compon
 	if err != nil {
 		return "", err
 	}
+	c, f := t.ComponentMaps[componentName]
+	if f && c.FlattenValues {
+		globals, ok := mergedVals["global"].(map[string]any)
+		if !ok {
+			return "", fmt.Errorf("global value isn't a map")
+		}
+		components, ok := mergedVals[c.ToHelmValuesTreeRoot].(map[string]any)
+		if !ok {
+			return "", fmt.Errorf("component value isn't a map")
+		}
+		finalVals := map[string]any{}
```

Comment:
you could reset mergedVals and reuse that instead of creating finalVals since globals and components hold a ref to the data.